tests/input_converter/test_converter.py

```diff
@@ -847,173 +840,236 @@ def test_convert_links_to_component(self, local_study_w_links: Study, lib_id: st
+
         self._validate_component(
-            instance, "process_nb_units_max", expected_path, expected_values
+            component, "nb_units_max", expected_path, expected_values
         )
 
-    @pytest.mark.parametrize("direction", ["forward", "backward"])
-    def test_nb_units_max_variation(
+    def nb_units_max_variation(
         self,
         local_study_w_thermal: Study,
-        create_csv_from_constant_value: Callable[..., None],
         direction: Literal["forward"] | Literal["backward"],
     ):
         """
         Tests nb_units_max_variation_forward and nb_units_max_variation_backward processing.
         """
-        instance, expected_path = self._setup_test(
-            local_study_w_thermal, f"nb_units_max_variation_{direction}"
-        )
-        modulation_timeseries = (
-            instance.study_path / "input" / "thermal" / "prepro" / "fr" / "gaz"
-        )
-        series_path = (
-            instance.study_path / "input" / "thermal" / "series" / "fr" / "gaz"
+
+        tdp = self._setup_preprocessing_thermal(local_study_w_thermal)
+        expected_path = (
+            local_study_w_thermal.service.config.study_path
+            / "input"
+            / "thermal"
+            / "series"
+            / "fr"
+            / "gaz"
+            / f"nb_units_max_variation_{direction}.txt"
         )
-        create_csv_from_constant_value(modulation_timeseries, "modulation", 840, 4)
-        create_csv_from_constant_value(series_path, "series", 840)
-        instance.process_nb_units_max()
-        nb_units_max_output = pd.read_csv(
-            instance.series_path / "nb_units_max.txt", header=None
+        tdp.generate_component_parameter("nb_units_max")
+
+        variation_component = tdp.generate_component_parameter(
+            f"nb_units_max_variation_{direction}"
         )
 
-        variation_component = getattr(
-            instance, f"process_nb_units_max_variation_{direction}"
-        )()
         current_df = pd.read_csv(variation_component.value + ".txt", header=None)
 
-        assert current_df[0][0] == max(
-            0, nb_units_max_output[0][167] - nb_units_max_output[0][0]
-        )
-        assert current_df[0][3] == max(
-            0, nb_units_max_output[0][2] - nb_units_max_output[0][3]
-        )
-        assert current_df[0][168] == max(
-            0, nb_units_max_output[0][335] - nb_units_max_output[0][168]
+        nb_units_max_output = pd.read_csv(
+            tdp.series_path / "nb_units_max.txt", header=None
         )
-        assert variation_component.value == str(expected_path)
 
-    def test_nb_units_max_variation_forward(
-        self,
-        local_study_w_thermal: Study,
-        create_csv_from_constant_value: Callable[..., None],
-    ):
-        self.test_nb_units_max_variation(
-            local_study_w_thermal, create_csv_from_constant_value, direction="forward"
-        )
+        diff = nb_units_max_output.shift(-1) - nb_units_max_output
```

Comment:
Well I am not convinced by this, as you are in fact redoing what is in the core of the function `_compute_nb_units_max_variation` (slightly differently, which does not in fact give really the expected df). As it seems difficult to generate the whole expected_df, for clarity of the test, I would come back the the asserts made in the previous version of the code (on some particular elements)